lib/backend/dynamo/dynamodbbk.go

```diff
@@ -92,6 +92,9 @@ type Config struct {
 	// WriteTargetValue is the ratio of consumed write capacity to provisioned
 	// capacity. Required to be set if auto scaling is enabled.
 	WriteTargetValue float64 `json:"write_target_value,omitempty"`
+
+	// OnDemand sets on-demand capacity to the DynamoDB tables
+	OnDemand *bool `json:"on_demand,omitempty"`
```

Comment:
Let's change this to a `billing_mode: (provisioned|on-demand)` enum as discussed in https://github.com/gravitational/teleport/pull/28672#discussion_r1253201715.